### PR TITLE
FastDDS update with PKCS#11 support

### PR DESCRIPTION
**Pull request by maseabunikie**

### CODE REVIEW

File: builder/packaging/build.sh [hunk 1]
```diff
@@ -129,7 +129,13 @@ function build_process {
 		step build_underlay_deps /main_ws/src
 	fi
 
-	/packaging/package.sh
+	if [[ ! -e package.xml ]]; then
+		pushd $(dirname $(find -name package.xml))
```
Comment (joonas-fi) [inline builder/packaging/build.sh:133]: lol this is clever :laughing:

File: builder/packaging/build.sh [hunk 2]
```diff
@@ -129,7 +129,13 @@ function build_process {
 		step build_underlay_deps /main_ws/src
 	fi
 
-	/packaging/package.sh
+	if [[ ! -e package.xml ]]; then
```
Comment (joonas-fi) [inline builder/packaging/build.sh:132]: I think it'd be good to have comment for this branch (telling that this is a rare case) with something like this:
```bash
# some projects (currently one: mesh_com) don't have package.xml in project root dir so temporarily change to that dir for package.sh to work
```

**Review by lassilaiho — APPROVED** [review]

File: builder/packaging/rosdep.yaml [hunk 3]
```diff
@@ -12,13 +12,13 @@ dynamicEDT3D:
...
 fastrtps:
-    ubuntu: ros-galactic-fastrtps=2.3.4*
+    ubuntu: ros-galactic-fastrtps=2.5.0*
 fastrtps_cmake_module:
     ubuntu: ros-galactic-fastrtps-cmake-module=1.2.1*
 fog_msgs:
     ubuntu: ros-galactic-fog-msgs=0.0.8*
 foonathan_memory_vendor:
-    ubuntu: ros-galactic-foonathan-memory-vendor=1.2.0*
+    ubuntu: ros-galactic-foonathan-memory-vendor=1.1.0*
```
Comment (joonas-fi) [inline builder/packaging/rosdep.yaml:21]: This required a downgrade?

**Review by joonas-fi — APPROVED** [review]
LGTM, only one minor idea about adding a code comment, but not absolutely necessary

Good work! :+1:

File: builder/packaging/rosdep.yaml (hunk 3 above)
Comment (maseabunikie) [reply builder/packaging/rosdep.yaml:21]: I think that there was a typo in the previous version.

File: builder/packaging/build.sh (hunk 2 above)
Comment (maseabunikie) [reply builder/packaging/build.sh:132]: Good idea. I will add this comment.

File: builder/packaging/rosdep.yaml (hunk 3 above)
Comment (maseabunikie) [reply builder/packaging/rosdep.yaml:21]: I have double checked and it is actually a downgrade. When this work started there was version 1.1.0 but now we already have a newer version available.

**Pull request merged**